packages/sdk/src/across/lpFeeCalculator.ts

```diff
@@ -0,0 +1,38 @@
+import assert from "assert";
+import { Provider, Block } from "@ethersproject/providers";
+import { ethers, BigNumber } from "ethers";
+import { bridgePool } from "../clients";
+import { RATE_MODELS } from "./constants";
+import { BigNumberish } from "./utils";
+import { calculateRealizedLpFeePct } from "./feeCalculator";
+import { exists, BlockFinder } from "../utils";
+
+export default class LpFeeCalculator {
+  private blockFinder: BlockFinder<Block>;
+  constructor(private provider: Provider) {
+    this.blockFinder = new BlockFinder<Block>(provider.getBlock.bind(provider));
+  }
+  async getLpFeePct(tokenAddress: string, bridgePoolAddress: string, amount: BigNumberish, timestamp?: number) {
+    const rateModel = RATE_MODELS[ethers.utils.getAddress(tokenAddress)];
+    assert(rateModel, "No rate model for token: " + tokenAddress);
+
+    amount = BigNumber.from(amount);
+    assert(amount.gt(0), "Amount must be greater than 0");
+
+    const { blockFinder, provider } = this;
+
+    const bridgePoolInstance = bridgePool.connect(bridgePoolAddress, provider);
+
+    const targetBlock = exists(timestamp)
```

Comment:
random aside: I really like that you have an `exists` exported method in utils, we should use this everywhere including in the bots!

Reply:
Actually, I notice that the return type of `exists` is `value`, shouldn't it be `boolean`?

Reply:
This was actually copied from some of matts code, which he has a similar function defined somewhere. If you are referring to this part ` value is T`, this is meant to let typescript know that the value is of type T, if the function returns true ( or something like that).  It does actually return a boolean, which looks really weird now thaat you bring it up

Reply:
@nicholaspai `exists` is a generic typescript method, so the `value is T` is a typescripty way of telling the compiler that the input (value) is of type T (which is the input type without the possibility of null or undefined).

It can be used as a javascript shorthand for "is the thing I passed in not undefined/null", but the reason we wrote it this way was to help typescript infer types. Here's a dumb example that illustrates what this allows us to do:
```ts
const a: number | undefined | null = someFun();
if (exists(a)) {
  // This compiles, because typescript knows that a cannot be undefined or null in this branch.
  const b: number = a;
} else {
  // This compiles because we know that a _must be_ undefined or null in this branch.
  let c: undefined | null = a;
}
```

It feels a little stupid/obvious that `a !== null && a !== undefined` means that a is a number, but typescript often fails to infer this unless you define these special functions to help it differentiate the cases.